### PR TITLE
vaultwarden: 1.30.5 -> 1.31.0

### LINKED ISSUE #325679: Update request: vaultwarden 1.30.5 → 1.31.0
- Package name: **vaultwarden**
- Latest released version: **1.31.0** (**[GitHub release](https://github.com/dani-garcia/vaultwarden/releases/tag/1.31.0)**, **[commit changelog](https://github.com/dani-garcia/vaultwarden/compare/1.30.5...1.31.0)**)

- Current version on the unstable channel: 1.30.5
- Current version on the stable/release channel: 1.30.5

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**: @dotlambda @SuperSandro2000 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by wahjava**
## Description of changes

https://github.com/dani-garcia/vaultwarden/releases/tag/1.31.0

https://github.com/dani-garcia/bw_web_builds/releases/tag/v2024.5.1b

Fixes #325679

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

closes https://github.com/NixOS/nixpkgs/pull/325727

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jackyliu16** [comment]
You may want to add yourself as maintainers of this packages.

**Review by dotlambda — CHANGES_REQUESTED** [review]
The backup test fails. Is
```
cp: cannot copy a directory, '/var/lib/vaultwarden/backups', into itself, '/var/lib/vaultwarden/backups/backups'
```
expected? It also occurs when I run the test on master but the test doesn't fail.
cc @Scrumplex @caffineehacker

**Comment by Scrumplex** [comment]
> The backup test fails. Is
> 
> ```
> cp: cannot copy a directory, '/var/lib/vaultwarden/backups', into itself, '/var/lib/vaultwarden/backups/backups'
> ```
> 
> expected? It also occurs when I run the test on master but the test doesn't fail.

This was caused by #318933. For some reason the backup test has its backup directory set to `/var/lib/vaultwarden` which is now also the default directory for `stateVersion` >= 24.11. This should only affect the test unless people changed the default `/var/backup/vaultwarden` path.

**Comment by Scrumplex** [comment]
See #325861 for the backup script issue

**Comment by dotlambda** [comment]
> > 
> 
> This was caused by #318933. For some reason the backup test has its backup directory set to `/var/lib/vaultwarden` which is now also the default directory for `stateVersion` >= 24.11. This should only affect the test unless people changed the default `/var/backup/vaultwarden` path.

That doesn't explain the test succeeding on master but failing for this PR.

**Comment by SuperSandro2000** [comment]
> You may want to add yourself as maintainers of this packages.

No, we don't need more maintainers to defuse ownership just because one update wasn't done on time.

File: pkgs/tools/security/vaultwarden/default.nix [hunk 1]
```diff
@@ -42,7 +42,7 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/dani-garcia/vaultwarden";
     changelog = "https://github.com/dani-garcia/vaultwarden/releases/tag/${version}";
     license = licenses.agpl3Only;
-    maintainers = with maintainers; [ dotlambda SuperSandro2000 ];
+    maintainers = with maintainers; [ dotlambda SuperSandro2000 abbe ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/vaultwarden/default.nix:45]: Two are fine, we don't need another one.

**Comment by Scrumplex** [comment]
> That doesn't explain the test succeeding on master but failing for this PR.

In my case the test succeeded on master but showed the same error you mentioned. The test was just faulty IMO

**Comment by dotlambda** [comment]
> In my case the test succeeded on master but showed the same error you mentioned. The test was just faulty IMO

The test succeeds on master for me too, but fails when I check out 1b8364eb8cfa.

**Comment by Scrumplex** [comment]
The test fails in this PR with this output:

```
Test "Check that backup exists" failed with error: "command `[ -f "/var/lib/vaultwarden/backups/rsa_key.pub.pem" ]` failed (exit code 1)"
```

Is it possible that this file doesn't exist anymore?

**Comment by dotlambda** [comment]
> Is it possible that this file doesn't exist anymore?

Indeed: https://github.com/dani-garcia/vaultwarden/pull/4085

**Comment by SuperSandro2000** [comment]
> > Is it possible that this file doesn't exist anymore?
> 
> Indeed: [dani-garcia/vaultwarden#4085](https://github.com/dani-garcia/vaultwarden/pull/4085)

Thanks for digging that up! When that fixes the test, I push the change to this PR.

I also applied the other review comments.

**Comment by Scrumplex** [comment]
@ofborg test vaultwarden.sqlite-backup

**Review by Scrumplex — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Review by dotlambda — APPROVED** [review]

**Pull request merged**